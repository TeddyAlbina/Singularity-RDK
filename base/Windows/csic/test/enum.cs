enum Color { Red, Green=2, Blue };
class test {
	static Color y = Color.Green;
	static test() {}
	public static void Main() {
		Color x = y;
		System.Console.WriteLine("{0}={1}", x, (int)x);
	}
}
