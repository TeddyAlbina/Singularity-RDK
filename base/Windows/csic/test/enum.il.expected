.assembly 'enum' { }
.assembly extern 'mscorlib' {
.ver 1:0:5000:0
.publickeytoken = ( b7 7a 5c 56 19 34 e0 89 )
}
.class auto ansi serializable sealed 'Color' extends [mscorlib]System.Enum
{
.field public static literal valuetype Color Red = int32(0)
.field public static literal valuetype Color Green = int32(2)
.field public static literal valuetype Color Blue = int32(3)
.field public rtspecialname specialname int32 value__
} // end of enumeration Color
.class 'test'
	extends [mscorlib]System.Object
{
.field static private value class Color 'y'
.method hidebysig specialname rtspecialname static private void .cctor() {
ldc.i4 2
stsfld value class Color 'test'::'y'
ret
.maxstack 1
} // end of constructor test.test
.method hidebysig public static void 'Main'() {
.entrypoint
.locals init ([0]value class Color 't0')
ldsfld value class Color 'test'::'y'
stloc 0 // x
ldstr "{0}={1}"
ldloc 0 // x
box value class Color
ldloc 0 // x
box int32
call  void [mscorlib]System.Console::'WriteLine'(string,object,object)
ret
.maxstack 3
} // end of method test.Main
.method hidebysig specialname rtspecialname public void .ctor() {
ldarg.0
call instance void [mscorlib]System.Object::.ctor()
ret
.maxstack 1
} // end of constructor test.test
} // end of class test
